nemo/utils/exp_manager.py

```diff
@@ -307,9 +308,21 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
             exp_dir, name, and version.
     """
     # Add rank information to logger
-    # Note: trainer.global_rank and trainer.is_global_zero are not set until trainer.fit, so have to hack around it
     local_rank = int(os.environ.get("LOCAL_RANK", 0))
-    global_rank = trainer.node_rank * trainer.num_devices + local_rank
+
+    # the exp manager calls some operations that require explicit
+    # synchronization, therefore we need to initialize the process
+    # group to initiate a barrier
+    if parallel_state.is_unitialized():
```

Comment:
just check if torch.distrib is init here, don't need megatron.core

Reply:
also make sure there's more than one gpu here

Reply:
fixed https://github.com/NVIDIA/NeMo/pull/7497/commits/07ff8ee4e1c6bf3ac75ba68ece1716aeec2ba921#diff-cacb4d3cf39593d980bcfcd9c7ed4b9a342b9565c5447511758daf1f02e0f60fR316